assert/http_assertions.go

```diff
@@ -90,6 +90,29 @@ func HTTPError(t TestingT, handler http.HandlerFunc, method, url string, values
 	return isErrorCode
 }
 
+// HTTPStatusCode asserts that a specified handler returns a specified status code.
+//
+//  assert.HTTPStatusCode(t, myHandler, "GET", "/notImplemented", nil, 501)
+//
+// Returns whether the assertion was successful (true) or not (false).
+func HTTPStatusCode(t TestingT, handler http.HandlerFunc, method, url string, values url.Values, statuscode int, msgAndArgs ...interface{}) bool {
```

Comment:
Would it not make sense to change the `method` argument (in both this and the other functions) into an enum? This would break backwards compatibility so it would have to wait till v2 but in my opinion makes a lot more sense as there are only a finite amount of verbs supported in HTTP standards.

Reply:
The comment you made makes sense to me for a v2... however, that feels like a form of scope creep to me?

May I suggest a middle ground for this?

My suggestion would be to merge this PR so people can start using it.

I can make a new PR with your change that can be labelled v2 or something or merged into a v2 branch. (didn't check if there already is one)

Reply:
Good point. It would make this PR too big and lose focus. 👍